libs/components/packages/src/schematics/migrations/update-7/ag-grid-28.schematic.ts

```diff
@@ -0,0 +1,195 @@
+      tree,
+      `${ENT_MODULE}all-modules`
+    );
+    const agGridDependencyEnt = getPackageJsonDependency(tree, AG_GRID_ENT);
+    let type = NodeDependencyType.Default;
+    const overwrite = true;
+    const version = UPDATE_TO_VERSION;
+    if (!agGridAllModules && !agGridDependency) {
+      // AG Grid is not installed, so we don't need to do anything.
+      return;
+    }
+
+    // Determine if there are any community packages that need to be updated.
+    if (
+      agGridDependency &&
+      semver.lt(agGridDependency.version, UPDATE_LT_VERSION)
+    ) {
+      type = agGridDependency.type;
+      needsPackageUpdated = true;
+    } else if (!agGridDependency && agGridAllModules) {
+      type = agGridAllModules.type;
+      needsPackageUpdated = true;
+    }
+    if (needsPackageUpdated) {
+      [AG_GRID, 'ag-grid-angular'].forEach((name) => {
+        addPackageJsonDependency(tree, { name, overwrite, type, version });
+      });
+    }
+    if (agGridAllModules) {
+      removePackageJsonDependency(tree, `${ANY_MODULE}all-modules`);
+      removePackageJsonDependency(tree, `${ANY_MODULE}angular`);
+    }
+
+    // Determine if there are any enterprise packages that need to be updated.
+    if (
+      agGridDependencyEnt &&
+      semver.lt(agGridDependencyEnt.version, UPDATE_LT_VERSION)
+    ) {
+      type = agGridDependencyEnt.type;
+      needsEntPackageUpdated = true;
+    } else if (!agGridDependencyEnt && agGridAllModulesEnt) {
+      type = agGridAllModulesEnt.type;
+      needsEntPackageUpdated = true;
+    }
+    if (needsEntPackageUpdated) {
+      addPackageJsonDependency(tree, {
+        name: AG_GRID_ENT,
+        overwrite,
+        type,
+        version,
+      });
+    }
+    if (agGridAllModulesEnt) {
+      removePackageJsonDependency(tree, `${ENT_MODULE}all-modules`);
+    }
+
+    // Install the new packages.
+    if (
+      needsPackageUpdated ||
+      needsEntPackageUpdated ||
+      agGridAllModules ||
+      agGridAllModulesEnt
+    ) {
+      context.addTask(new NodePackageInstallTask());
+    }
+
+    const ignoreList = tree.exists('.gitignore')
+      ? tree.readText('.gitignore')
+      : '';
+    const ig = ignore().add(ignoreList);
+
+    tree.visit((filePath) => {
+      // If the file is not a TypeScript file or if it is ignored by git, we can skip it.
+      if (!filePath.endsWith('.ts') || ig.ignores(relative('/', filePath))) {
+        return;
+      }
+      const content = tree.readText(filePath);
+      let updatedContent = content;
+
+      // Prompt the user to moderate the use of AG Grid modules
+      if (
+        updatedContent.includes(ANY_MODULE) ||
+        updatedContent.includes(ENT_MODULE)
+      ) {
+        warnOnce(`\n
+        AG Grid recommends not mixing module and package imports.
+        https://ag-grid.com/angular-data-grid/modules/\n\n`);
+      }
+      // Replace `@ag-grid-community/all-modules` with `ag-grid-community`
```

Comment:
We already have a rule that handles the replacement of import statements. Maybe it would work for your needs?
https://github.com/blackbaud/skyux/blob/main/libs/components/packages/src/schematics/rules/replace-typescript-imports.ts